src/core/linalg/src/sparse/4C_linalg_map.hpp

```diff
@@ -0,0 +1,158 @@
+// This file is part of 4C multiphysics licensed under the
+// GNU Lesser General Public License v3.0 or later.
+//
+// See the LICENSE.md file in the top-level for license information.
+//
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#ifndef FOUR_C_LINALG_MAP_HPP
+#define FOUR_C_LINALG_MAP_HPP
+
+
+#include "4C_config.hpp"
+
+#include "4C_comm_mpi_utils.hpp"
+
+#include <Epetra_Map.h>
+#include <Epetra_MpiComm.h>
+#include <mpi.h>
+
+#include <memory>
+
+FOUR_C_NAMESPACE_OPEN
+
+// Do not lint the file for identifier names, since the naming of the Wrapper functions follow the
+// naming of the Epetra_Map
+
+// NOLINTBEGIN(readability-identifier-naming)
+
+namespace Core::LinAlg
+{
+
+  class Map
+  {
+   public:
+    //! Creates a Epetra_Map object based on Epetra_Comm
+    Map(int NumGlobalElements, int IndexBase, const Epetra_Comm& Comm);
+
+    Map(int NumGlobalElements, int IndexBase, const MPI_Comm& Comm);
+
+    Map(int NumGlobalElements, int NumMyElements, int IndexBase, const Epetra_Comm& Comm);
+
+    Map(int NumGlobalElements, int NumMyElements, int IndexBase, const MPI_Comm& Comm);
+
+    Map(int NumGlobalElements, int NumMyElements, const int* MyGlobalElements, int IndexBase,
+        const Epetra_Comm& Comm);
+
+    Map(int NumGlobalElements, int NumMyElements, const int* MyGlobalElements, int IndexBase,
+        const MPI_Comm& Comm);
+
+    Map(const Map& Source);
+
+    /// Copy constructor from Epetra_Map
+    explicit Map(const Epetra_Map& Source);
+
+    /// Copy constructor from Epetra_Map
+    explicit Map(const Epetra_BlockMap& Source);
+
+    Map& operator=(const Map& other);
+    Map(Map& other);
```

Comment:
That's uncommon. The copy constructor is already declared in the usual manner above.